flare/src/main/java/io/pantheon/flare/adapters/FirebaseAnalyticsAdapter.kt

```diff
@@ -13,26 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.pantheon.flare
+package io.pantheon.flare.adapters
 
 import android.content.Context
 import com.google.firebase.analytics.FirebaseAnalytics
+import io.pantheon.flare.BundleJSONConverter
 import org.json.JSONObject
 
 class FirebaseAnalyticsAdapter : AnalyticsAdapter<FirebaseAnalytics>() {
 
     private lateinit var firebaseAnalytics: FirebaseAnalytics
 
     override fun initialize(block: FirebaseAnalytics?.() -> Unit): AnalyticsAdapter<FirebaseAnalytics> {
-        FlareProvider.flareContext?.let { context: Context ->
+        context?.let { context: Context ->
             firebaseAnalytics = FirebaseAnalytics.getInstance(context)
+            block(firebaseAnalytics)
         }
-        block(firebaseAnalytics)
         return this
     }
 
     override fun logEvent(eventName: String, eventMap: HashMap<String?, Any?>) {
         require(::firebaseAnalytics.isInitialized) { Exception("Flare: FirebaseAnalytics not initialized") }
```

Comment:
Can we shift this check to base class as well?

Reply:
Done!